### PR TITLE
Make images clickable

### LINKED ISSUE #57: Make images in visual novel media clickable
Images there should be clickable. On click zoom in or something like that should occur.
image

### LINKED ISSUE #67: Implement image carousel for visual novel media images
You can check [this](https://codesandbox.io/s/dd8vn?file=/src/ChakraCarousel.js) example for reference

### PR DESCRIPTION
**Pull request by Moltenship**
Make images in visual novel media clickable #57

### CODE REVIEW

File: src/features/VisualNovel/pages/VisualNovelPage/VisualNovelPage.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const VisualNovelPage: FC = () => {
       {data && (
         <div className={cls.page}>
           <header className={cls.header}>
-            <Image src={data.image as string} alt={data.title} className={cls.image} />
+            <Image src={data.image ?? undefined} loading="eager" alt={data.title} className={cls.image} />
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/VisualNovelPage/VisualNovelPage.tsx]: Should we render image if there are no source?

Comment (NightOfNights) [reply src/features/VisualNovel/pages/VisualNovelPage/VisualNovelPage.tsx]: Maybe add placeholder

File: src/components/ImagePreviewDialog/ImagePreviewDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+/** Dialog for previewing images. */
+const ImagePreviewDialogComponent: VFC<Props> = ({ image, onClose, isOpen }) => (
+  <Modal isOpen={isOpen} onClose={onClose} size="2xl" isCentered>
+    <ModalOverlay />
+    <ModalContent>
+      <Image src={image} />
+    </ModalContent>
+  </Modal>
```
Comment (NightOfNights) [inline src/components/ImagePreviewDialog/ImagePreviewDialog.tsx:23]: I think it should be image carousel, instead of single image

Comment (NightOfNights) [reply src/components/ImagePreviewDialog/ImagePreviewDialog.tsx:23]: Maybe implement it somehow like [here](https://codesandbox.io/s/dd8vn?file=/src/ChakraCarousel.js)

Comment (NightOfNights) [reply src/components/ImagePreviewDialog/ImagePreviewDialog.tsx:23]: Never mind, let's implement in another [task](https://github.com/vndbre/vndbre/issues/67)

File: src/features/VisualNovel/pages/MediaPage/MediaPage.module.css [hunk 3]
```diff
@@ -7,4 +7,5 @@
 .image {
   border-radius: 8px;
   width: 320px;
```
Comment (moetazaneta) [inline src/features/VisualNovel/pages/MediaPage/MediaPage.module.css]: You should use design tokens.

**Review by moetazaneta — CHANGES_REQUESTED** [review]
use design tokens.

File: src/features/VisualNovel/pages/MediaPage/MediaPage.tsx [hunk 4]
```diff
@@ -12,23 +13,54 @@ import cls from './MediaPage.module.css';
...
+
   const images = data && (
     <div className={cls.list}>
-      {data.screens.filter(filterPredicate).map(screen => <Image key={screen.image} src={screen.image} className={cls.image} />)}
+      {data.screens.filter(filterPredicate).map((screen, idx) => (
+        <Image
+          onClick={() => handleClickOnImage(idx)}
+          key={screen.image}
+          src={screen.image}
+          className={cls.image}
```
Comment (moetazaneta) [inline src/features/VisualNovel/pages/MediaPage/MediaPage.tsx]: Remind you to use design tokens or chakra props!

File: src/features/VisualNovel/pages/MediaPage/MediaPage.tsx [hunk 5]
```diff
@@ -12,23 +13,54 @@ import cls from './MediaPage.module.css';
...
+        />
+      ))}
+      <Viewer
+        visible={isVisible}
+        images={data.screens.filter(filterPredicate).map(screen => ({ src: screen.image }))}
```
Comment (moetazaneta) [inline src/features/VisualNovel/pages/MediaPage/MediaPage.tsx]: Please, be mindful about code duplication.

**Review by moetazaneta — APPROVED** [review]

**Pull request merged**